core/schema/host.go

```diff
@@ -151,7 +151,7 @@ func (s *hostSchema) Install(srv *dagql.Server) {
 				dagql.Arg("exclude").Doc(`Exclude artifacts that match the given pattern (e.g., ["node_modules/", ".git*"]).`),
 				dagql.Arg("include").Doc(`Include only artifacts that match the given pattern (e.g., ["app/", "package.*"]).`),
 				dagql.Arg("noCache").Doc(`If true, the directory will always be reloaded from the host.`),
-				dagql.Arg("noGitAutoIgnore").Doc(`Don't apply .gitignore filter rules inside the directory`),
+				dagql.Arg("gitignore").Doc(`Apply .gitignore filter rules inside the directory`),
```

Comment:
The big change here is this.

I want to make this opt-in - we made a breaking change accidentally, and I think it's worth backing out of that. Since this API cannot be called by modules, it's outside module-compat so we should be fine for that.

This makes more sense with the approach we're going for now; we only need to add it for module loading, instead of needing to opt out for flag loading, and for context directories, and tests, etc.